__tests__/chat.test.js

```diff
@@ -0,0 +1,263 @@
+      await bob.chat.send(channel, message)
+      let messages = await alice.chat.read(channel)
+      expect(messages[0]).toHaveProperty('unread', true)
+      messages = await alice.chat.read(channel)
+      expect(messages[0]).toHaveProperty('unread', false)
+
+      // Now let's peek. Messages should remain unread on subsequent reads.
+      await bob.chat.send(channel, message)
+      messages = await alice.chat.read(channel, {peek: true})
+      expect(messages[0]).toHaveProperty('unread', true)
+      messages = await alice.chat.read(channel)
+      expect(messages[0]).toHaveProperty('unread', true)
+    })
+
+    it('Throws an error if given an invalid channel', async () => {
+      expect(alice.chat.read(invalidChannel)).rejects.toThrowError()
+    })
+  })
+
+  describe('Chat send', () => {
+    it('Sends a message to a certain channel and returns an empty promise', async () => {
+      await alice.chat.send(channel, message)
+
+      const messages = await alice.chat.read(channel, {
+        peek: true,
+      })
+      expect(messages[0].sender.username).toEqual(alice.myInfo().username)
+      expect(messages[0].content.text.body).toEqual(message.body)
+    })
+
+    it('Throws an error if given an invalid channel', async () => {
+      expect(alice.chat.send(invalidChannel, message)).rejects.toThrowError()
+    })
+
+    it('Throws an error if given an invalid message', async () => {
+      expect(alice.chat.send(channel, invalidMessage)).rejects.toThrowError()
+    })
+  })
+
+  describe('Chat delete', () => {
+    it('Deletes a message to a certain channel and returns an empty promise', async () => {
+      await alice.chat.send(channel, message)
+
+      // Send a message
+      const messages = await alice.chat.read(channel, {
+        peek: true,
+      })
+      expect(messages[0].sender.username).toEqual(alice.myInfo().username)
+      expect(messages[0].content.text.body).toEqual(message.body)
+
+      const {id} = messages[0]
+      await alice.chat.delete(channel, id)
+
+      // Send a message
+      const newMessages = await alice.chat.read(channel, {
+        peek: true,
+      })
+      expect(newMessages[0].id).toEqual(id + 1)
+      expect(newMessages[0].content.delete.messageIDs).toContain(id)
+      expect(newMessages[0].content.delete.messageIDs).toHaveLength(1)
+      expect(newMessages[1].id).toEqual(id - 1)
+    })
+
+    it('Throws an error if given an invalid channel', async () => {
+      await alice.chat.send(channel, message)
+      const messages = await alice.chat.read(channel, {
+        peek: true,
+      })
+      const {id} = messages[0]
+      expect(alice.chat.delete(invalidChannel, id)).rejects.toThrowError()
+    })
+
+    it('Throws an error if given an invalid id', async () => {
+      expect(alice.chat.send(channel, -1)).rejects.toThrowError()
+    })
+
+    it('Throws an error if it cannot delete the message (e.g., someone else wrote it)', async () => {
```

Comment:
This error currently fails; I wrote it before I realized that anyone can actually delete any message in a conversation. Just wanted to check before I remove it as I know that feature isn't in the GUI.

Reply:
hmm, it was my understanding that in team chats, admins could delete from anyone. and in DM's people could only delete their own. But this is a one-on-one chat, outside a team, and alice is deleting bob's message?

Reply:
I'll make a core ticket for this.